src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -180,7 +180,8 @@ class InfiniteScroll extends PureComponent {
         this.setState(
           { beginPage: nextBeginPage, endPage: nextEndPage },
           () => {
-            if (onMore && nextEndPage === lastPage) {
+            // Trigger onMore function as nextEndPage value is incremented as user scrolls through
+            if (onMore && nextEndPage > endPage && endPage <= lastPage) {
```

Comment:
This is calling `onMore` too often. It should only be called when reaching the lastPage. To clarify...
Say `step={10}` and `items` has 20 items in it. Initially, InfiniteScroll will render the first 10 items. As the user scrolls down and reaches the end of those 10 items, it will render the next 10 items that it already has from the items provided. When the user reaches the end of those 20 items, then `onMore` will be called. `onMore` is there so that the caller can provide more data in `items`. As long as InfiniteScroll has items that it can step through, there is no need to call `onMore`.

Reply:
From existing issues and discussions, I was under impression that `onMore` is supposed to be called based on `step` value. From your example, I was expecting `onMore` to get called as it renders every 10 items. Based on my understanding, logic has been changed to handle `onMore` function as user scroll through. But looks like this is not what is expected to happen. I think I need to include another condition to check whether `items` length changed since last time it loaded items to avoid calling `onMore` too often as it renders the items equal to `step` value .

@ShimiSun Isn't this what was expected for the test case that we discussed over slack? Were you expecting `onMore` to get called two times when there are `30` items and `step` value is `10`?

Reply:
Sorry for the long overdue response. @mrebb, thank you for your work here, Eric's clarification got me on the right track of understanding the expected functionality. I will close this PR, we can discuss further offline if still needed. Thank you for the collaboration!